src/wireguard_interface.rs

```diff
@@ -10,6 +10,11 @@ pub trait WireguardInterfaceApi {
     /// Assigns IP address to an existing interface.
     fn assign_address(&self, address: &IpAddrMask) -> Result<(), WireguardInterfaceError>;
 
+    /// Add peer routing, basically a copy of `wg-quick up <if_name>` routing.
+    /// Extracts all uniques allowed ips from [Peer](crate::Peer) slice and add routing for every
+    /// address.
+    fn configure_peers_routing(&self, peers: &[Peer]) -> Result<(), WireguardInterfaceError>;
```

Comment:
Still not a fan of this name. Maybe consider `add_peer_routing`. At least change peers to peer to be consistent. Also the docstring could use some editing (a summary line and grammar fixes).